### PR TITLE
Truncates inputs logged when a node errors

### PR DESCRIPTION
**Pull request by skrawcz**
This was requested by a user, because we could overwhelm a log system. This should suffice for now. We can always modify tweak this later.

## Changes
- graph_functions.py error handling code

## How I tested this
 - locally

```
********************************************************************************
> spend_per_signup [my_functions.spend_per_signup()] encountered an error      <
> Node inputs:
{'signups': '0      1\n1     10\n2     50\n3    100\n4    200\n5    ...',
 'spend': '0    10\n1    10\n2    20\n3    40\n4    40\n5    50\ndt...'}
********************************************************************************
```

## Checklist

- [ ] PR has an informative and human-readable title (this will be pulled into the release notes)
- [ ] Changes are limited to a single goal (no scope creep)
- [ ] Code passed the pre-commit check & code is left cleaner/nicer than when first encountered.
- [ ] Any _change_ in functionality is tested
- [ ] New functions are documented (with a description, list of inputs, and expected output)
- [ ] Placeholder code is flagged / future TODOs are captured in comments
- [ ] Project documentation has been updated if adding/changing functionality.

### CODE REVIEW

File: hamilton/execution/graph_functions.py [hunk 1]
```diff
@@ -95,6 +96,29 @@ def combine_config_and_inputs(config: Dict[str, Any], inputs: Dict[str, Any]) ->
...
+    for k, v in kwargs.items():
+        item_repr = repr(v)
+        if len(item_repr) > 50:
+            item_repr = item_repr[:50] + "..."
+        else:
+            item_repr = v
+        inputs[k] = item_repr
+    input_string = pp.pformat(inputs)
+    if len(input_string) > 1000:
+        input_string = input_string[:80] + "..."
```
Comment (elijahbenizzy) [inline hamilton/execution/graph_functions.py:118]: Why 80 versus 1000? Shouldn't we truncate it to 1000?

File: hamilton/execution/graph_functions.py [hunk 2]
```diff
@@ -95,6 +96,29 @@ def combine_config_and_inputs(config: Dict[str, Any], inputs: Dict[str, Any]) ->
     return {**config, **inputs}
 
 
+def create_input_string(kwargs: dict) -> str:
+    """THis is a utility function to create a string representation of the inputs to a function.
```
Comment (elijahbenizzy) [inline hamilton/execution/graph_functions.py:100]: This

**Review by elijahbenizzy — COMMENTED** [review]
Looks good -- we don't have a unit tests for the overall, non-item-level truncation, but we should probably just fix that (and maybe add a test?)

File: hamilton/execution/graph_functions.py (hunk 1 above)
Comment (skrawcz) [reply hamilton/execution/graph_functions.py:118]: oops, yeah missed that.

**Comment by skrawcz** [comment]
> Looks good -- we don't have a unit tests for the overall, non-item-level truncation, but we should probably just fix that (and maybe add a test?)

? yes we do.

**Review by elijahbenizzy — APPROVED** [review]

**Pull request merged**